packages/eas-cli/src/build/metadata.ts

```diff
@@ -29,6 +36,17 @@ export async function collectMetadata<T extends Platform>(
     credentialsSource?: CredentialsSource.LOCAL | CredentialsSource.REMOTE;
   }
 ): Promise<Metadata> {
+  const [channel, releaseChannel] = await Promise.all([
+    resolveChannel(ctx),
+    resolveReleaseChannel(ctx),
+  ]);
+  if (channel && releaseChannel) {
+    Log.warn(
+      `Both channel: "${channel}" and releaseChannel: "${releaseChannel}" are defined. Defaulting to channel.`
```

Comment:
- This just resolving metadata, it should not trigger any warnings
- I'm not sure if it's correct behaviour e.g.
  - if you have release channel specified in Expo.plist and channel in eas.json it should add it should use `channel`
  - if you have channel specified in Expo.plist and release channel in eas.json it should add it should use `releaseChannel`
  
  I suggest creating one resolve function, sth like this
```
if (!installed)  ...
if (buildProfile.channel) ...
if (buildProfile.releasChannel) ...
if (channelForNativeConfig) ...
if (releaseChannelFromNativeConfig) ...
```

Reply:
OK, I like that.